ConsoleHabitTracker.kraven88/Menu.cs

```diff
@@ -0,0 +1,296 @@
+            return false;
+
+        else if (int.TryParse(select, out int number) && number > 0 && number <= habits.Count)
+        {
+            var habit = habits.Where(x => x.Id == number).First();
+            return HabitMenu(habit);
+        }
+        else 
+            return true;
+
+    }
+    
+    private bool SetDailyGoal(Habit habit)
+    {
+        HeaderText(habit.Name);
+        
+        Console.WriteLine($"{habit.Name} current daily goal is {habit.CurrectGoal} {habit.UnitOfMeasurement}");
+
+        var newGoal = AskForInteger("Enter new daily goal (0 to leave unchanged): ");
+
+        if (newGoal == 0) return true;
+        else
+        {
+            db.SaveNewGoal(habit, newGoal);
+            habit.CurrectGoal = newGoal;
+        }
+
+        return true;
+        
+    }
+    
+    private bool UpdateDailyProgress(Habit habit)
+    {
+        HeaderText(habit.Name);
+        var today = DateOnly.FromDateTime(DateTime.Now).ToString("dd.MM.yyyy");
+
+        var newProgress = AskForInteger($"Please enter the number of ({habit.UnitOfMeasurement}) since your last update: ");
+        if (newProgress == 0) return true;
+        else
+        {
+            db.SaveProgress(habit, today, newProgress);
+            Console.WriteLine("Progress succesfuly updated.");
+            Console.ReadKey();
+            return true;
+        }
+    }
+    
+    private bool ViewCurrentProgress(Habit habit)
+    {
+        HeaderText(habit.Name);
+        var today = DateOnly.FromDateTime(DateTime.Now).ToString("dd.MM.yyyy");
+
+        //db.SaveProgress(habit, today, 0);   // This ensures that, when selected, Current Progress will always include current day data
+        try
+        { 
+            habit.ProgressList = db.LoadCurrentProgress(habit);
+        }
+        catch (ArgumentException ex)
+        {
+            Console.WriteLine($"Oops! Something went wrong: {ex.Message}");
+            Console.ReadKey();
+            return true;
+        }
+
+        var current = habit.ProgressList.LastOrDefault();
+        Console.WriteLine($"Todays progress for {habit.Name}: [ {habit.ProgressList.First().Quantity} / {habit.ProgressList.First().DailyGoal} ] {habit.UnitOfMeasurement}.");
+        Console.ReadLine();
+
+        return true;
+    }
+
+    private bool ViewEntireProgress(Habit habit)
+    {
+        HeaderText(habit.Name);
+        try
+        {
+            habit.ProgressList = db.LoadAllProgress(habit);
+        }
+        catch (ArgumentException ex)
+        {
+            Console.WriteLine($"Oops! Something went wrong: {ex.Message}");
+            Console.ReadKey();
+            return true;
+        }
+        var progress = habit.ProgressList.OrderByDescending(x => x.Date);
+
+        foreach (var day in progress)
+            Console.WriteLine($" Day: {day.Date}, [ {day.Quantity} / {day.DailyGoal} ] {habit.UnitOfMeasurement}");
+
+        Console.ReadKey();
+
+        return true;
+    }
```

Comment:
Your program didn't build because of the missing curly bracket.

Reply:
Seems like I accidentally removed it when I was cleaning code just before PR. Silly me.